rclpy/test/test_node.py

```diff
@@ -838,6 +838,119 @@ def test_node_set_parameters_rejection(self):
         self.assertIsInstance(result[0], SetParametersResult)
         self.assertFalse(result[0].successful)
 
+    def reject_parameter_callback_1(self, parameter_list):
+        rejected_parameters = (param for param in parameter_list if 'reject1' in param.name)
+        return SetParametersResult(successful=(not any(rejected_parameters)))
+
+    def reject_parameter_callback_2(self, parameter_list):
+        rejected_parameters = (param for param in parameter_list if 'reject2' in param.name)
+        return SetParametersResult(successful=(not any(rejected_parameters)))
+
+    def reject_parameter_callback_3(self, parameter_list):
+        rejected_parameters = (param for param in parameter_list if 'reject3' in param.name)
+        return SetParametersResult(successful=(not any(rejected_parameters)))
+
+    def test_node_set_paramters_rejection_list(self):
+        # Declare a new paramters and set a list of callbacks so that it's rejected when set.
+        reject_list_parameter_tuple1 = [
+            (
+                'reject1',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                'non_reject',
+                True,
+                ParameterDescriptor()
+            ),
+            (
+                'non_reject',
+                True,
+                ParameterDescriptor()
+            )
+        ]
+
+        self.node.declare_parameters('', reject_list_parameter_tuple1)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_1)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_2)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_3)
+
+        result = self.node.set_parameters(
+            [
+                Parameter(
+                    name=reject_list_parameter_tuple1[0][0],
+                    value=reject_list_parameter_tuple1[0][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[1][0],
+                    value=reject_list_parameter_tuple1[1][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple1[2][0],
+                    value=reject_list_parameter_tuple1[2][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertFalse(result[0].successful)
+
+    def test_node_add_on_set_paramter_callback(self):
+        # Add callbacks to the list of callbacks.
+
+        callbacks = [
+            self.reject_parameter_callback,
+            self.reject_parameter_callback_1,
+            self.reject_parameter_callback_2,
+            self.reject_parameter_callback_3
+        ]
+        for callback in callbacks:
+            self.node.add_on_set_parameters_callback(callback)
```

Comment:
I feel like this is insufficiently testing this. It should verify that the callbacks are in the list.

Reply:
Should I implement this like the way I have implemented remove test function?

Reply:
@suab321321 I left some suggested changes for you.

Reply:
@allenh1 sir I am unable to understand the changes you suggested me..could you please explain a bit?

Reply:
Absolutely. I left explanations for each below.